frontend/src/packages/sites/sites-api.ts

```diff
@@ -44,14 +46,28 @@ class SitesApi extends HttpApi {
     return await response.json<SiteCreateResponseDto>();
   }
 
-  public async getByProjectId(
-    projectId: number,
-  ): Promise<SiteGetAllResponseDto> {
+  private isQueryParametersExist(
+    queryParameters: SitesFilterQueryDto,
+  ): boolean {
+    {
+      return Object.values(queryParameters).some((element) => element !== '');
+    }
+  }
+
+  public async getByProjectId({
+    projectId,
+    queryParameters,
+  }: SiteGetByProjectParametersDto): Promise<SiteGetAllResponseDto> {
+    const searchParameters = this.isQueryParametersExist(queryParameters)
```

Comment:
Not sure if you need a helper for forming search params, check out how it was done on `getProjects`

Reply:
http://localhost:3000/api/v1/sites/project/2?name= 
even when we don`t have any search parameters query is present in this case, it`s ok?